django/db/models/sql/query.py

```diff
@@ -328,6 +328,8 @@ def clone(self):
             obj.subq_aliases = self.subq_aliases.copy()
         obj.used_aliases = self.used_aliases.copy()
         obj._filtered_relations = self._filtered_relations.copy()
+        if self.combined_queries:
+            obj.combined_queries = copy.deepcopy(obj.combined_queries)
```

Comment:
I'd try calling `clone` recursively instead.

```suggestion
            obj.combined_queries = tuple(query.clone() for query in obj.combined_queries)
```

Reply:
If I'm not mistaken we won't need this at all after the changes suggested below.

Reply:
@ChetanKhanna yeah I agree that it's likely unnecessary given `tuples` are immutable and we don't alter `combined_queries` except during expression resolving.